src/Abstractions/TaskOrchestrationContext.cs

```diff
@@ -260,10 +260,9 @@ public async Task<T> WaitForExternalEvent<T>(string eventName, TimeSpan timeout)
     /// <typeparam name="TResult">
     /// The type into which to deserialize the sub-orchestrator's output.
     /// </typeparam>
-    /// <inheritdoc cref="CallSubOrchestratorAsync(TaskName, string?, object?, TaskOptions?)"/>
+    /// <inheritdoc cref="CallSubOrchestratorAsync(TaskName, object?, TaskOptions?)"/>
     public abstract Task<TResult> CallSubOrchestratorAsync<TResult>(
         TaskName orchestratorName,
-        string? instanceId = null,
         object? input = null,
         TaskOptions? options = null);
```

Comment:
Have you considered simply changing `TaskOptions` to `OrchestrationOptions` here? I wonder if that would make setting the instance ID more discoverable.

Reply:
The intent here was to allow re-use of the same `TaskOptions` object between activity invocation and sub-orchestration invocation, with appending instance ID as the less-often use case.

There are 3 ways I see to make it more discoverable:

1. [**Current**] - have a `.WithInstanceId(string? instanceId)` method on `TaskOptions`. - this shows you _can_ use instanceId if you want to.
2. Add an overload which takes in `OrchestrationOptions` and passes to the `TaskOptions` overload.
    - A little counter-intuitive from implementers accessing `InstanceId`, but it saves object creation/copying in what I assume is the more common scenario (supplying retry options, no instance Id - just use `TaskOptions`)
3. Like option 2, but reverse: `TaskOptions` overload passes to `OrchestrationOptions` overload.
    - Implementations have access to `InstanceId` without any special logic. However, using `TaskOptions` will always lead to an object copy.

Reply:
After further though, I kept with the current style. The reason is the options parameter style is meant to be flexible, different implementors can define their own options. For example, maybe durable functions extension wants to define a `FunctionsTaskOptions` with special functions-only behavior. Going with option 3 would prevent this behavior as custom types would always be lost as it is converted to `SubOrchestrationOptions`. This leaves option 1 or 2, and option 2 would not be a breaking change, so we can add that later if we get customer feedback about discoverability concerns.

Reply:
I see your point about reusing the same `TaskOptions` object between activity and sub-orchestrations - I agree there is value there. I think I also see your point about defining custom options types. I do worry about discoverability, but custom instance IDs probably enough of a special case that it's not going to hurt anyone to have to find this simple extension method. I'm good to keep the design as-is.